jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -885,35 +895,68 @@ protected void startContext() throws Exception
 
     public void contextDestroyed() throws Exception
     {
-        MultiException ex = new MultiException();
-        ServletContextEvent event = new ServletContextEvent(_scontext);
-        Collections.reverse(_destroyServletContextListeners);
-        for (ServletContextListener listener : _destroyServletContextListeners)
+        switch (_contextStatus)
```

Comment:
perhaps a comment saying where this can be called from.
Also can we move the method to be after contextInitialized... I like order to match natural sequence.